src/vs/workbench/contrib/preferences/browser/keybindingsEditor.ts

```diff
@@ -135,10 +135,16 @@ export class KeybindingsEditor extends BaseEditor implements IKeybindingsEditor
 		this.createBody(keybindingsEditorElement);
 	}
 
-	setInput(input: KeybindingsEditorInput, options: EditorOptions | undefined, token: CancellationToken): Promise<void> {
+	setInput(input: KeybindingsEditorInput, options: SettingsEditorOptions | undefined, token: CancellationToken): Promise<void> {
 		this.keybindingsEditorContextKey.set(true);
-		return super.setInput(input, options, token)
-			.then(() => this.render(!!(options && options.preserveFocus)));
+		return super.setInput(input, options, token).then(() => {
+			this.render(!!(options && options.preserveFocus)).then(() => {
+				options = options || SettingsEditorOptions.create({});
+				if (options.query) {
+					this.searchWidget.setValue(options.query);
+				}
+			});
+		});
```

Comment:
I do not think you need these changes if you use `IKeybindingEditorSearchOptions`

Reply:
Thanks, I didn't notice search value could be provided this way! After adoping this option there's no need to change `KeybindingsEditor`'s implementation.